tokens/usecase/tokens_usecase_test.go

```diff
@@ -539,5 +542,418 @@ func (s *TokensUseCaseTestSuite) TestGetMinPoolLiquidityCap() {
+			},
+			expectedResult: "",
+			expectedError:  fmt.Errorf("chain denom not found in chain registry"),
+		},
+		{
+			name:       "Invalid type - not a string",
+			chainDenom: "invalidType",
+			coingeckoIdsMap: map[string]any{
+				"invalidType": 123,
+			},
+			expectedResult: "",
+			expectedError:  fmt.Errorf("coingecko id for chain denom (invalidType) is not of type string"),
+		},
+	}
+
+	for _, tt := range testcases {
+		s.Run(tt.name, func() {
+			usecase := tokensusecase.NewTokensUsecase(nil, 0, nil)
+			for k, v := range tt.coingeckoIdsMap {
+				usecase.SetCoingeckoIDs(k, v)
+			}
+
+			result, err := usecase.GetCoingeckoIdByChainDenom(tt.chainDenom)
+			if tt.expectedError != nil {
+				s.Require().EqualError(err, tt.expectedError.Error())
+			} else {
+				s.Require().NoError(err)
+			}
+			s.Require().Equal(tt.expectedResult, result)
+		})
+	}
+}
+
+// MockTokenLoader is a mock implementation of TokenLoader.
+type MockTokenLoader struct {
+	loadTokens tokensusecase.LoadTokensFunc
+	callCount  int
+	err        error
+}
+
+func (m *MockTokenLoader) FetchAndUpdateTokens() error {
+	m.callCount++
+	return m.err
+}
+
+func (m *MockTokenLoader) CallCount() int {
+	return m.callCount
+}
+
+// TestUpdateAssetsAtHeightIntervalSync tests the async update of assets at height interval.
+func (s *TokensUseCaseTestSuite) TestUpdateAssetsAtHeightIntervalSync() {
+	testcases := []struct {
+		name              string
+		height            uint64
+		interval          uint64
+		loader            *MockTokenLoader
+		expectedCallCount int
+		expectedErr       error
+	}{
+		{
+			name:              "Height matches interval, no error",
+			height:            10,
+			interval:          10,
+			loader:            &MockTokenLoader{},
+			expectedCallCount: 1,
+			expectedErr:       nil,
+		},
+		{
+			name:              "Height does not match interval",
+			height:            9,
+			interval:          10,
+			loader:            &MockTokenLoader{},
+			expectedCallCount: 0,
+			expectedErr:       nil,
+		},
+		{
+			name:     "Height matches interval, with error",
+			height:   20,
+			interval: 10,
+			loader: &MockTokenLoader{
+				err: fmt.Errorf("mock error"),
+			},
+			expectedCallCount: 1,
+			expectedErr:       fmt.Errorf("mock error"),
+		},
+		{
+			name:              "No loader, no error",
+			height:            20,
+			interval:          10,
+			loader:            nil,
+			expectedCallCount: 0,
+			expectedErr:       nil,
+		},
+	}
+
+	for _, tt := range testcases {
+		s.Run(tt.name, func() {
+			usecase := tokensusecase.NewTokensUsecase(
+				nil,
+				int(tt.interval),
+				noOpLogger,
+			)
+
+			if tt.loader != nil {
+				usecase.SetTokenRegistryLoader(tt.loader) // set loader if non nil, otherwise *MockTokenLoader is not nil
+			}
+
+			err := usecase.UpdateAssetsAtHeightIntervalSync(tt.height)
+			if err != nil && err.Error() != tt.expectedErr.Error() {
+				s.Assert().Failf("expected error: %v, got: %v", tt.expectedErr.Error(), err)
```

Comment:
Let's please use the following pattern for stylistic consistency and readability
```

if tt.expectedErr != nil {
   s.Require().Error(err)
   s.Require().ErrorIs(tt.expectedErr, err)
   return 
}

s.Require().NoError(err)

...
```

Reply:
Good catch, updated, thanks!